sktime/forecasting/compose/_skforecast_reduce.py

```diff
@@ -0,0 +1,452 @@
+    ) -> None:
+        self.regressor = regressor
+        self.lags = lags
+        self.transformer_y = transformer_y
+        self.transformer_exog = transformer_exog
+        self.weight_func = weight_func
+        self.differentiation = differentiation
+        self.fit_kwargs = fit_kwargs
+        self.binner_kwargs = binner_kwargs
+
+        super().__init__()
+
+        self._regressor = None
+        self._forecaster = None
+        self._transformer_y = None
+        self._transformer_exog = None
+
+        self._clone_estimators()
+
+    def _clone_estimators(self: "SkforecastAutoreg"):
+        """Clone the regressor and transformers."""
+        from sklearn.base import clone
+
+        self._regressor = clone(self.regressor)
+
+        if self.transformer_y:
+            self._transformer_y = clone(self.transformer_y)
+
+        if self.transformer_exog:
+            self._transformer_exog = clone(self.transformer_exog)
+
+    def _create_forecaster(self: "SkforecastAutoreg"):
+        """Create ``skforecast.ForecasterAutoreg.ForecasterAutoreg`` model."""
+        from skforecast.ForecasterAutoreg import ForecasterAutoreg
+
+        return ForecasterAutoreg(
+            self._regressor,
+            self.lags,
+            transformer_y=self._transformer_y,
+            transformer_exog=self._transformer_exog,
+            weight_func=self.weight_func,
+            differentiation=self.differentiation,
+            fit_kwargs=self.fit_kwargs,
+            binner_kwargs=self.binner_kwargs,
+        )
+
+    @staticmethod
+    def _coerce_column_names(X: Optional[pd.DataFrame]):
+        if X is None:
+            return None
+
+        return X.rename(columns=lambda column_name: str(column_name))
+
+    @staticmethod
+    def _coerce_int_to_range_index(df):
+        new_df = df.copy(deep=True)
+        start = new_df.index[0]
+        stop = new_df.index[-1] + 1
+        if len(new_df.index) == 1:
+            step = 1
+        else:
+            step = new_df.index[1] - start
+
+        new_index = pd.RangeIndex(start, stop, step)
+        np.testing.assert_array_equal(new_df.index, new_index)
+        new_df.index = new_index
+        return new_df
+
+    @staticmethod
+    def _coerce_period_to_datetime_index(df):
+        new_df = df.copy(deep=True)
+        period_freq = new_df.index.freq
+        # converting the period index to the timestamp at the 'end' of the period
+        # Example:        Period                   Datetime
+        #          '2024-01-12 00:00' --> '2024-01-12 00:59:59.999999999'
+        #          '2024-01-12 01:00' --> '2024-01-12 01:59:59.999999999'
+        new_df.index = new_df.index.to_timestamp(how="e")
+        new_df.index.freq = period_freq
+        return new_df
+
+    def _make_index_compatible(self, df, input_var):
+        if df is None:
+            return None
+
+        if isinstance(df.index, (pd.RangeIndex, pd.DatetimeIndex)):
+            return df
+
+        new_df = df
```

Comment:
Is this assignment used?

Reply:
No, it is not necessary now. I've removed it.